commands/command_lock.go

```diff
@@ -91,6 +92,9 @@ func lockPath(file string) (string, error) {
 	}
 
 	path = filepath.ToSlash(path)
+	if strings.HasPrefix(path, "../") {
+		return "", fmt.Errorf("lfs: unable to canonicalize path %q", path)
```

Comment:
Can the `../` component occur elsewhere in the path, or can it only be the first element?

Reply:
I believe it can only be the first element. We produce a path relative to the repository root. If the path is under the repository root (by whatever metric), it won't contain a `..` component, since we'll have canonicalized both paths already. If it isn't, the first component will contain `..`, since we'll need to leave the repository root.

I considered checking in the middle of the path as well, but it's not strictly necessary and there isn't a simple way to check the entire path without also catching weird paths like `/home/foo/path-ending-in-two-dots../`, which I feel confident someone, somewhere, would complain about.